tangelo/toolboxes/ansatz_generator/puccd.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2021 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module defines the pUCCD ansatz class. The molecular FermionOperator is
+expected to be converted to a BosonOperator (electrons in pairs). Single bosonic
+excitations (corresponding to double fermion excitations) form the ansatz. Those
+excitations are transformed into a quantum circuit via Givens rotations.
+
+Ref:
+    - V.E. Elfving, M. Millaruelo, J.A. Gámez, and C. Gogolin.
+        Phys. Rev. A 103, 032605 (2021).
+"""
+
+import itertools
+import numpy as np
+
+from tangelo.linq import Circuit
+from tangelo.toolboxes.ansatz_generator import Ansatz
+from tangelo.toolboxes.ansatz_generator.ansatz_utils import givens_gate
+from tangelo.toolboxes.qubit_mappings.statevector_mapping import vector_to_circuit
+
+
+class pUCCD(Ansatz):
+    """This class implements the pUCCD ansatz. Only closed-shell molecules are
+    supported.
```

Comment:
I need you to remove the ambiguity

1. If this reflects the current state of the code and open-shell support will be implemented in the future: remove this and let your `NotImplementedError` handle it. We dont want to have to update documentatino every time code change.

2. Otherwise state that pUCCD was not designed to support open-shell systems. Your error then becomes a `ValueError`: the value passed by the users are not adequate. Your current error mean "it's not yet implemented but it will be later"